launcher/ui/dialogs/BlockedModsDialog.cpp

```diff
@@ -186,8 +185,24 @@ void BlockedModsDialog::setupWatch()
 {
     const QString downloadsFolder = APPLICATION->settings()->get("DownloadsDir").toString();
     const QString modsFolder = APPLICATION->settings()->get("CentralModsDir").toString();
-    m_watcher.addPath(downloadsFolder);
-    m_watcher.addPath(modsFolder);
+    const bool downloadsFolderWatchRecursive = APPLICATION->settings()->get("DownloadsDirWatchRecursive").toBool();
+    watchPath(downloadsFolder, downloadsFolderWatchRecursive);
+    watchPath(modsFolder, true);
+}
+
+void BlockedModsDialog::watchPath(QString path, bool watch_subdirectories)
+{
+    qDebug() << "[Blocked Mods Dialog] Adding Watch Path:" << path;
+    m_watcher.addPath(path);
+
+    if (!watch_subdirectories)
+        return;
+
+    QDirIterator it(path, QDir::Filter::Dirs | QDir::Filter::NoDotAndDotDot, QDirIterator::NoIteratorFlags);
+    while (it.hasNext()) {
+        QString dir_path = it.next();
+        watchPath(dir_path, watch_subdirectories);
+    }
```

Comment:
i think this needs a bit of logic to not recurse into itself with symlinks
![symbolic link in mods folder, linking to the mods folder itself, being followed recursively](https://user-images.githubusercontent.com/9145768/229371020-c6c14484-18c1-43fe-bdca-8243ad78ce81.png)

Reply:
now that's just evil >:(

please confirm this is fixed with the latest :P

Reply:
The recursion displayed before is fixed, though the `mods` folder now got added twice:
![two mods folders are present: one with a relative path, and one with an absolute path](https://user-images.githubusercontent.com/9145768/229464893-590aa3b9-d4ab-4ad7-b75f-5654c98d29bc.png)
not a big issue though

Reply:
ok, now it *really* can't duplicate a path to watch!

Reply:
now it doesn't work as intended though :P

Commit 5ce7874:
![image](https://user-images.githubusercontent.com/9145768/230492647-b7f2d223-ba05-4de0-87c3-eda03d8ab0a1.png)

Commit 62c5982:
![image](https://user-images.githubusercontent.com/9145768/230492744-51fc0215-eab0-42b7-817e-ef30b1188907.png)

I suppose in the line `auto to_watch_path = to_watch.canonicalPath();`, the call to `canonicalPath()` needs to be replaced by `canonicalFilePath()` so as to not consume the last element of the path.

Reply:
.....
*sigh*

clearly I forgot to rebuild somehow before I ran my tests.